vision_datasets/image_object_detection/detection_as_kvp_dataset.py

```diff
@@ -0,0 +1,113 @@
+        f"{BBOXES_KEY}": {
+            "type": "array",
+            "description": "Bounding boxes of objects in the image",
+            "items": {
+                "type": "string",
+                "description": "Class name that the box belongs to",
+                "classes": {},
+                "includeGrounding": True
+            }
+        }
+    }
+}
+
+
+class DetectionAsKeyValuePairDataset(VisionDataset):
+    """Dataset class that access Detection datset as KeyValuePair dataset."""
+
+    def __init__(self, detection_dataset: VisionDataset):
+        """
+        Initializes an instance of the ClassificationAsKeyValuePairDataset class.
+        Args:
+            detection_dataset (VisionDataset): The detection dataset to convert to key-value pair dataset.
+        """
+
+        if detection_dataset is None or detection_dataset.dataset_info.type not in {DatasetTypes.IMAGE_OBJECT_DETECTION}:
+            raise ValueError("DetectionAsKeyValuePairDataset only supports Image Object Detection datasets.")
+
+        # Generate schema and update dataset info
+        detection_dataset = deepcopy(detection_dataset)
+
+        dataset_info_dict = detection_dataset.dataset_info.__dict__
+        dataset_info_dict["type"] = DatasetTypes.KEY_VALUE_PAIR.name.lower()
+        self.class_names = [c.name for c in detection_dataset.dataset_manifest.categories]
+        self.class_id_to_names = {c.id: c.name for c in detection_dataset.dataset_manifest.categories}
+        self.img_id_to_pos = {x.id: i for i, x in enumerate(detection_dataset.dataset_manifest.images)}
+
+        schema = self.construct_schema(self.class_names)
+        # Update dataset_info with schema
+        dataset_info = KeyValuePairDatasetInfo({**dataset_info_dict, "schema": schema})
+
+        # Construct KeyValuePairDatasetManifest
+        annotations = []
+        for id, img in enumerate(detection_dataset.dataset_manifest.images, 1):
+            bboxes = [box.label_data for box in img.labels]
+
+            kvp_label_data = self.construct_kvp_label_data(bboxes)
+            img_ids = [self.img_id_to_pos[img.id]]  # 0-based index
+            kvp_annotation = KeyValuePairLabelManifest(id, img_ids, label_data=kvp_label_data)
+
+            # KVPDatasetManifest expects img.labels to be empty. Labels are instead stored in KVP annotation
+            img.labels = []
+            annotations.append(kvp_annotation)
+
+        dataset_manifest = KeyValuePairDatasetManifest(detection_dataset.dataset_manifest.images, annotations, schema, additional_info=detection_dataset.dataset_manifest.additional_info)
+        super().__init__(dataset_info, dataset_manifest, dataset_resources=detection_dataset.dataset_resources)
+
+    def construct_schema(self, class_names: List[str]) -> Dict[str, Any]:
+        schema: Dict[str, Any] = BASE_DETECTION_SCHEMA  # initialize with base schema
+        schema["fieldSchema"][f"{BBOXES_KEY}"]["items"]["classes"] = {c: {} for c in class_names}
```

Comment:
could we add a description if it is single-class? I have found that otherwise the schema confuses the current gpt4o. You can add something like:
{c: {'description': f'Always output {c} as the class.'}}

Reply:
Added description. Please let me know if it too long now :)

Reply:
Sorry for not being clear, I meant to add a default description if there is only one class in this task. If there are multiple classes, we won't need the default description.